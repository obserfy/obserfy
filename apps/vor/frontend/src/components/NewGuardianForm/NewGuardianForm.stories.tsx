import { Meta, Story } from "@storybook/react"
import NewGuardianForm, { NewGuardianFormProps } from "./NewGuardianForm"

export default {
  title: "Core/NewGuardianForm",
  component: NewGuardianForm,
  parameters: {
    componentSubtitle: "Just a simple NewGuardianForm",
  },
} as Meta

const Template: Story<NewGuardianFormProps> = (args) => (
  <NewGuardianForm {...args} />
)

export const Default = Template.bind({})
Default.args = {}
