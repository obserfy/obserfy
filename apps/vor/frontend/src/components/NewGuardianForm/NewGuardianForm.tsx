import { t } from "@lingui/macro"
import { FC, useState } from "react"
import { Box } from "theme-ui"
import Input from "../Input/Input"
import TextArea from "../TextArea/TextArea"

interface NewGuardian {
  name: string
  email: string
  phone: string
  note: string
  address: string
}
export interface NewGuardianFormProps {
  newGuardian: NewGuardian
  onChange: (guardian: NewGuardian) => void
}
const NewGuardianForm: FC<NewGuardianFormProps> = ({
  newGuardian,
  onChange,
}) => {
  const { name, note, email, phone, address } = newGuardian

  const handleChange = (newData: Partial<NewGuardian>) => {
    onChange({
      ...newGuardian,
      ...newData,
    })
  }

  return (
    <Box p={3}>
      <Input
        value={name}
        mb={2}
        label={t`Guardian Name`}
        sx={{ width: "100%" }}
        onChange={(e) => handleChange({ name: e.target.value })}
      />
      <Input
        type="email"
        value={email}
        mb={2}
        label={t`Email`}
        sx={{ width: "100%" }}
        onChange={(e) => handleChange({ email: e.target.value })}
      />
      <Input
        type="phone"
        value={phone}
        mb={3}
        label={t`Phone`}
        sx={{ width: "100%" }}
        onChange={(e) => handleChange({ phone: e.target.value })}
      />
      <Input
        value={address}
        mb={3}
        label={t`Address`}
        sx={{ width: "100%" }}
        onChange={(e) => handleChange({ address: e.target.value })}
      />
      <TextArea
        mb={3}
        label={t`Note`}
        value={note}
        onChange={(e) => handleChange({ note: e.target.value })}
      />
    </Box>
  )
}

export const useNewGuardianFormState = () =>
  useState({
    email: "",
    name: "",
    phone: "",
    note: "",
    address: "",
  })

export default NewGuardianForm
