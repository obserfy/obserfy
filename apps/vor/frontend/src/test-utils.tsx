/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
