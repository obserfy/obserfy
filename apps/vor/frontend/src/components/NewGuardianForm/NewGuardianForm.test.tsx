import { render } from "../../test-utils"
import NewGuardianForm from "./NewGuardianForm"

describe("NewGuardianForm", () => {
  it("should render correctly", () => {
    const { container } = render(
      <NewGuardianForm
        newGuardian={{
          email: "",
          name: "",
          phone: "",
          note: "",
          address: "",
        }}
        onChange={() => {}}
      />
    )
    expect(container).toMatchSnapshot()
  })
})
