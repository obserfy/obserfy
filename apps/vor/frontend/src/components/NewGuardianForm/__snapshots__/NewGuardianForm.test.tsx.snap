// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewGuardianForm should render correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: 16px;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: var(--theme-ui-colors-textMediumEmphasis);
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: var(--theme-ui-colors-textMediumEmphasis);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-bottom: 4px;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  width: auto;
  border-color: var(--theme-ui-colors-border);
  background-color: var(--theme-ui-colors-surface);
  font-size: 16px;
  padding: 12px;
  color: var(--theme-ui-colors-text);
  margin-bottom: 8px;
  --theme-ui-input-autofill-bg: var(--theme-ui-colors-background);
  width: 100%;
  width: 100%;
}

.emotion-5:autofill,
.emotion-5:autofill:hover,
.emotion-5:autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-5:autofill:first-line,
.emotion-5:autofill:hover:first-line,
.emotion-5:autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-5:-webkit-autofill,
.emotion-5:-webkit-autofill:hover,
.emotion-5:-webkit-autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-5:-webkit-autofill:first-line,
.emotion-5:-webkit-autofill:hover:first-line,
.emotion-5:-webkit-autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-5:hover,
.emotion-5:focus {
  border-color: var(--theme-ui-colors-primary);
}

.emotion-5::-webkit-input-placeholder {
  font-style: italic;
}

.emotion-5::-moz-placeholder {
  font-style: italic;
}

.emotion-5:-ms-input-placeholder {
  font-style: italic;
}

.emotion-5::placeholder {
  font-style: italic;
}

.emotion-5:disabled {
  opacity: 0.68;
  border-color: var(--theme-ui-colors-border);
  color: var(--theme-ui-colors-textDisabled);
}

.emotion-13 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  width: auto;
  border-color: var(--theme-ui-colors-border);
  background-color: var(--theme-ui-colors-surface);
  font-size: 16px;
  padding: 12px;
  color: var(--theme-ui-colors-text);
  margin-bottom: 16px;
  --theme-ui-input-autofill-bg: var(--theme-ui-colors-background);
  width: 100%;
  width: 100%;
}

.emotion-13:autofill,
.emotion-13:autofill:hover,
.emotion-13:autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-13:autofill:first-line,
.emotion-13:autofill:hover:first-line,
.emotion-13:autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-13:-webkit-autofill,
.emotion-13:-webkit-autofill:hover,
.emotion-13:-webkit-autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-13:-webkit-autofill:first-line,
.emotion-13:-webkit-autofill:hover:first-line,
.emotion-13:-webkit-autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-13:hover,
.emotion-13:focus {
  border-color: var(--theme-ui-colors-primary);
}

.emotion-13::-webkit-input-placeholder {
  font-style: italic;
}

.emotion-13::-moz-placeholder {
  font-style: italic;
}

.emotion-13:-ms-input-placeholder {
  font-style: italic;
}

.emotion-13::placeholder {
  font-style: italic;
}

.emotion-13:disabled {
  opacity: 0.68;
  border-color: var(--theme-ui-colors-border);
  color: var(--theme-ui-colors-textDisabled);
}

.emotion-18 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: var(--theme-ui-colors-textMediumEmphasis);
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-20 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  border-color: var(--theme-ui-colors-border);
  background-color: var(--theme-ui-colors-surface);
  font-size: 16px;
  padding: 16px;
  height: 180px;
  resize: vertical;
  color: var(--theme-ui-colors-text);
  margin-bottom: 16px;
}

.emotion-20:hover,
.emotion-20:focus {
  border-color: var(--theme-ui-colors-primary);
}

.emotion-20::-webkit-input-placeholder {
  font-style: italic;
  line-height: 2rem;
}

.emotion-20::-moz-placeholder {
  font-style: italic;
  line-height: 2rem;
}

.emotion-20:-ms-input-placeholder {
  font-style: italic;
  line-height: 2rem;
}

.emotion-20::placeholder {
  font-style: italic;
  line-height: 2rem;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <label
        class="emotion-2"
      >
        <div
          class="emotion-3"
        >
          Guardian Name
        </div>
        <div
          class="emotion-4"
        >
          <input
            class="emotion-5"
            value=""
          />
        </div>
      </label>
      <label
        class="emotion-2"
      >
        <div
          class="emotion-3"
        >
          Email
        </div>
        <div
          class="emotion-4"
        >
          <input
            class="emotion-5"
            type="email"
            value=""
          />
        </div>
      </label>
      <label
        class="emotion-2"
      >
        <div
          class="emotion-3"
        >
          Phone
        </div>
        <div
          class="emotion-4"
        >
          <input
            class="emotion-13"
            type="phone"
            value=""
          />
        </div>
      </label>
      <label
        class="emotion-2"
      >
        <div
          class="emotion-3"
        >
          Address
        </div>
        <div
          class="emotion-4"
        >
          <input
            class="emotion-13"
            value=""
          />
        </div>
      </label>
      <label
        class="emotion-18"
      >
        <div
          class="emotion-3"
        >
          Note
        </div>
        <textarea
          aria-label="Note"
          class="emotion-20"
        />
      </label>
    </div>
  </div>
</div>
`;
